packages/grafana-ui/src/components/Table/TableNG/Cells/PillCell.tsx

```diff
@@ -34,38 +34,49 @@ function createPills(pillValues: string[], field: Field, theme: GrafanaTheme2):
   });
 }
 
-export function PillCell({ value, field }: TableCellRendererProps) {
+export function PillCell({ rowIdx, field }: PillCellProps) {
   const styles = useStyles2(getStyles);
   const theme = useTheme2();
+  const value = field.values[rowIdx];
 
   const pills: Pill[] = useMemo(() => {
-    const pillValues = inferPills(String(value));
-    return createPills(pillValues, field, theme);
+    const pillValues = inferPills(value);
+    return pillValues.length > 0 ? createPills(pillValues, field, theme) : [];
   }, [value, field, theme]);
 
-  return pills.map((pill) => (
-    <span
-      key={pill.key}
-      className={styles.pill}
-      style={{
-        backgroundColor: pill.bgColor,
-        color: pill.color,
-        border: pill.bgColor === TRANSPARENT ? `1px solid ${theme.colors.border.strong}` : undefined,
-      }}
-    >
-      {pill.value}
-    </span>
-  ));
+  if (pills.length === 0) {
+    return null;
+  }
+
+  return (
+    <div className={styles.wrapper}>
+      {pills.map((pill) => (
+        <span
+          key={pill.key}
+          className={styles.pill}
+          style={{
+            backgroundColor: pill.bgColor,
+            color: pill.color,
+            border: pill.bgColor === TRANSPARENT ? `1px solid ${theme.colors.border.strong}` : undefined,
+          }}
+        >
+          {pill.value}
+        </span>
+      ))}
+    </div>
+  );
 }
 
 const SPLIT_RE = /\s*,\s*/;
 const TRANSPARENT = 'rgba(0,0,0,0)';
 
-export function inferPills(value: string): string[] {
-  if (value === '') {
+export function inferPills(rawValue: TableCellValue): string[] {
+  if (rawValue === '' || rawValue == null) {
```

Comment:
if the value was `null`, we were seeing a single pill with `null` as the value. added a test to track this case.